File: mobile/src/main/java/treehou/se/habit/main/MainContract.java
package treehou.se.habit.main;


import se.treehou.ng.ohcommunicator.connector.models.OHSitemap;
import treehou.se.habit.mvp.BasePresenter;
import treehou.se.habit.mvp.BaseView;

public interface MainContract {

    interface View extends BaseView<Presenter> {
        void openSitemaps();
        void openSitemaps(String defaultSitemap);
        void openSitemap(OHSitemap ohSitemap);
        void openControllers();
        void openServers();
        void openSettings();
        boolean hasOpenPage();
    }

    interface Presenter extends BasePresenter {
        void showSitemaps();
        void showSitemap(OHSitemap ohSitemap);
        void showControllers();
        void showServers();
        void showSettings();
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/main/MainContract.kt
package treehou.se.habit.ui.main


import se.treehou.ng.ohcommunicator.connector.models.OHSitemap
import treehou.se.habit.mvp.BasePresenter
import treehou.se.habit.mvp.BaseView

interface MainContract {

    interface View : BaseView<Presenter> {
        fun openSitemaps()
        fun openSitemaps(defaultSitemap: String)
        fun openSitemap(ohSitemap: OHSitemap)
        fun openControllers()
        fun openServers()
        fun openSettings()
        fun hasOpenPage(): Boolean
    }

    interface Presenter : BasePresenter {
        fun showSitemaps()
        fun showSitemap(ohSitemap: OHSitemap)
        fun showControllers()
        fun showServers()
        fun showSettings()
    }
}
